src/riot-rs-threads/src/lib.rs

```diff
@@ -300,6 +324,10 @@ pub fn sleep() {
 /// Wakes up a thread and adds it to the runqueue.
 ///
 /// Returns `false` if no paused thread exists for `thread_id`.
+///
+/// # Panics
+///
+/// Panics if `thread_id` is >= [`THREADS_NUMOF`].
```

Comment:
See comment above on `get_state`.  
The call to `set_state` could panic, but because we only do that if `get_state` returned a valid value, `wakeup` itself won't panic.